finally updating the rate limits docs to be more helpful (#206)
Finch-API/finch-docs: docs/Best-Practices/Handling-Rate-Limit-Errors.md

# Handling Rate Limit Errors

If you are experiencing rate limits, there are several ways to minimize the risk of hitting rate limits.

## Implement batched requests

> Minimize the risk of hitting rate limits by batching individual requests into a single request. Several of Finch’s endpoints are “batch” endpoints ([/individual](https://developer.tryfinch.com/docs/reference/9d6c83b09e205-individual), [/employment](https://developer.tryfinch.com/docs/reference/1ba5cdec4c979-employment), and [/pay-statment](https://developer.tryfinch.com/docs/reference/d5fd02c41e83a-pay-statement)). Batch endpoints allow a list of IDs to be sent in a single request. Finch will correspondingly return a single response with an array of objects equal to the IDs sent.

Reviewer: ```suggestion
Minimize the risk of hitting rate limits by batching individual requests into a single request. Several of Finch’s endpoints are “batch” endpoints ([/individual](https://developer.tryfinch.com/docs/reference/9d6c83b09e205-individual), [/employment](https://developer.tryfinch.com/docs/reference/1ba5cdec4c979-employment), and [/pay-statment](https://developer.tryfinch.com/docs/reference/d5fd02c41e83a-pay-statement)). These batch endpoints allow you to send a list of IDs in a single request and receive a response with an array of objects matching the IDs sent.
```

There is no limit to the amount of IDs that can be sent in a single request. Determine your optimal batch size for your use case when making batch requests to Finch APIs. By passing all the required IDs in a single batch request optimizes your API usage and minimizes the risk of hitting rate limits.

A batch request example is as follows:

```json
{
  "requests": [
    {
      "individual_id": "772b3c4f-d764-433d-bd69-ff8bbac33ffe"
    },
    {
      "individual_id": "a7a77065-0f68-418d-a85a-da24fb2139b7"
    },

>     .
>     .
>     .

Reviewer: ```suggestion
    ...
```

    {
      "individual_id": "84364585-c2ce-40aa-bcc0-666ac9577315"
    }
  ]
}
```

## Implement a Rate Limiter Enforcer

> Suppose the rate limit for a particular Finch `product` is 20 requests per minute. You can use the following JavaScript code example to enforce this rate limit quota in your application. The `RateLimiter` class implements a simple rate limiter that allows you to make requests up to the specified rate limit (20 requests per minute) and pauses further requests until the rate limit resets before resuming API requests. Alternatively to pausing, you can implement a “back-off and retry” strategy. The `request` method of the rate limiter is used to make API requests to Finch's endpoints, ensuring that you stay within the rate limit "bucket" quota for that endpoint. Simply initialize a new `RateLimiter` class for each Finch `product` endpoint being called.

Reviewer: ```suggestion
Suppose the rate limit for a particular Finch `product` is 20 requests per minute. You can use the following JavaScript code example to enforce this rate limit quota in your application. The `RateLimiter` class implements a simple rate limiter that allows you to make requests up to the specified rate limit (20 requests per minute) and pauses further requests until the rate limit resets . As an alternative, you can implement a “back off and retry” strategy. The `request` method of the rate limiter is used to make API requests to Finch's endpoints, ensuring that you stay within the rate limit "bucket" quota for that endpoint. Simply initialize a new `RateLimiter` class for each Finch `product` endpoint being called.
```

1. Token A makes 4 requests to the [/employer/company](https://developer.tryfinch.com/docs/reference/33162be1eed72-company) endpoint, 3 requests to the [/employer/directory](https://developer.tryfinch.com/docs/reference/12419c085fc0e-directory) endpoint, and 2 requests to the [/employer/payment](https://developer.tryfinch.com/docs/reference/b811fdc2542ca-payment) endpoint within a minute.
    - Access Token A rate limits
        Bucket | Capacity
        -------|-------------
        `company` | 4/4 (FULL)
        `directory` | 3/4
        `individual` | 0/4
        `employment` | 0/4
        `payment` | 2/2 (FULL)
        `pay-statement` | 0/2
    - Application-level rate limits
        Bucket | Capacity
        -------|-------------
        `company` | 4/20
        `directory` | 3/20
        `individual` | 0/20
        `employment` | 0/20
        `payment` | 2/12
        `pay-statement` | 0/12

> 1. Token B makes 5 requests to the [/employer/company](https://developer.tryfinch.com/docs/reference/33162be1eed72-company) endpoint, 3 requests to the [/employer/directory](https://developer.tryfinch.com/docs/reference/12419c085fc0e-directory) endpoint, and 2 requests to the [/employer/payment](https://developer.tryfinch.com/docs/reference/b811fdc2542ca-payment) endpoint within the same minute. Token B’s first 4 requests to the [/employer/company](https://developer.tryfinch.com/docs/reference/33162be1eed72-company) endpoint will succeed, but the 5th will fail and return a 429 rate limit error. Note: Only succeeded requests count towards the application level rate limits.

Reviewer: each number here is "1". not sure if that's intentional
